pytest_asyncio/plugin.py

```diff
@@ -320,6 +320,9 @@ def pytest_pycollect_makeitem(
     if not collector.funcnamefilter(name):
         return None
     _preprocess_async_fixtures(collector.config, _HOLDER)
+    if isinstance(obj, staticmethod):
+        # staticmethods need to be unwrapped.
+        obj = getattr(obj, "__func__", False)
```

Comment:
```suggestion
        obj = obj.__func__
```
`staticmethod` always has `__func__` attr by definition.